### ISSUE TITLE
Separate the model and tensorflow initialization from its application for predictions

### ISSUE THREAD

**Issue by toliwa** [Franck-Dernoncourt/NeuroNER#5]
Thanks for providing NeuroNER, it is very nice. I trained a custom model based on some BRAT annotations, and was able to use this model to predict annotations for a new document.

From a user perspective, it would be very useful to separate the model and tensorflow initialization from its application for annotation predictions. 

Currently, everything happens in the `main()` method in the `main.py` file. I am thinking of a use case where NeuroNER could provide a method like `predict_annotations(doc)`, where given a string, it would apply a loaded model and return the BRAT annotation format. Then NeuroNER could be easily integrated into other scientific workflows or pipelines. Are there any plans for such a direction?

EDIT:
In line `270` in `main.py` I found a similar method to what I had in mind:

```y_pred, y_true, output_filepaths = train.predict_labels(sess, model, transition_params_trained, parameters, dataset, epoch_number, stats_graph_folder, dataset_filepaths)```

Of course `sess` and the other objects are all created in `main()`. It seems to me that the `main()` method could be refactored into an initialization component that sets up and returns `sess`, `model` and other objects, and further subsequent evaluation and output components, to facilitate pipeline use, but this is just a guess from looking at the code.

**Comment by Franck-Dernoncourt** [comment]
Thanks for the suggestion. Separating the model initialization from the prediction sounds like a great idea and a much-needed improvement. Pull requests are welcome :-)

**Comment by toliwa** [comment]
In pull request #8 , I added a functionality to `main.py`, so that `parameters_filepath` and the `output_folder` can be given as (optional) command-line argparse or method arguments. This way, at least experiment specific parameter files and output folders can be used when calling NeuroNER in the shell or when calling the main method from Python.

I also wrote a unit test in `test_main.py`, which is really more of a sanity test to see if model outputs are being written to the desired output folder. It can be run with `python test_main.py`.

The next steps would probably be to de-couple the logical parts in the main method into smaller methods, so that one could unit test them. Then, the `predict_labels` and `prediction_step methods` in `train.py` could be wrapped up into a prediction only method, so that given some text input, the predictions are returned.

I really hope this could be done by somebody with more NeuroNER knowledge, this program seems really useful and it would be great to do some scientific pipelines with it.
